guillotina/json/deserialize_value.py

```diff
@@ -37,18 +40,32 @@ def schema_compatible(value, schema_or_field, context=None):
         raise ValueDeserializationError(schema_or_field, value, "Deserializer not found for field")
 
 
+def _optimized_lookup(value, field, context):
```

Comment:
Couldn’t we just extend schema_compatible with optimized_lookup code ?

Reply:
good point!

Reply:
Actually, no we can't. We still need top-level fields to be looked up for correct implementation. For example, patch fields.